src/Features/Core/Portable/PullMemberUp/Dialog/PullMemberUpWithDialogCodeAction.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.PullMemberUp;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal abstract partial class AbstractPullMemberUpRefactoringProvider
+    {
+        private class PullMemberUpWithDialogCodeAction : CodeActionWithOptions
+        {
+            private readonly ISymbol _selectedNodeSymbol;
+
+            private readonly Document _contextDocument;
+
+            private readonly IPullMemberUpOptionsService _service;
+
+            private readonly SemanticModel _semanticModel;
```

Comment:
do you need the semantic model?  you already have the _contextDocument.  Is this a semantic model for *another* document?  If so, that needs to be documented.